java-frontend/src/main/java/org/sonar/java/se/ExplodedGraphWalker.java

```diff
@@ -187,6 +190,7 @@ private void execute(MethodTree tree) {
     methodTree = tree;
     constraintManager = new ConstraintManager();
     workList = new LinkedList<>();
+    endOfExecutionPath = new HashSet<>();
```

Comment:
Why not use a set which preserve the order, like `LinkedHashSet`? It would allow to keep the current test working and keep predictability.